dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java

```diff
@@ -2322,9 +2322,20 @@ private Queries getQueries(Field field) {
                         .append(field.getFieldContentlet()).append("` != ");
             }
         } else {
-            whereField.append(field.getFieldContentlet()).append(" IS NOT NULL AND ").append(field.getFieldContentlet())
-                    .append(" != ");
+            whereField.append(field.getFieldContentlet()).append(" IS NOT NULL AND ");
+            if (!DbConnectionFactory.isMsSql()){
+                whereField.append(field.getFieldContentlet()).append(" != ");
+            } else {
+                //Specific logic for text(varchar) and textarea (text) fields
+                if (field.getFieldContentlet().contains("text_area")){
+                    whereField.append(" DATALENGTH (").append(field.getFieldContentlet()).append(")");
+                } else{
+                    whereField.append(field.getFieldContentlet()).append(" != ");
+                }
+            }
```

Comment:
I think this if else if else logic can be simplified a bit:
if(DbConnectionFactory.isMsSql() && field.getFieldContentlet().contains("text_area")) {
    whereField.append(" DATALENGTH (").append(field.getFieldContentlet()).append(")");
} else {
  whereField.append(field.getFieldContentlet()).append(" != ");
}